### PR TITLE
add option to deploy to single node

### PR DESCRIPTION
**Pull request by tommy-mor**
this

1. adds --singleNode option to every command in ./bin/rama-cluster.sh
2.  splits the terraform home directories into two, one for single, one for multi
3.  puts shared code/scripts in a common directory
4. updates readme with shared instructions

### CODE REVIEW

File: rama-cluster/multi/cluster.tf [hunk 1]
```diff
@@ -286,6 +291,7 @@ resource "aws_instance" "supervisor" {
 ###
 
 resource "null_resource" "zookeeper" {
+  #TODO commonify these things
```
Comment (nathanmarz) [inline rama-cluster/multi/cluster.tf]: don't leave stray TODOs

File: rama-cluster/single/start.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+sudo yum update -y
+
+echo "Starting zookeeper..." >> setup.log
+
+sudo cp zookeeper.service /etc/systemd/system &>> setup.log
+sudo systemctl start zookeeper.service &>> setup.log
+sudo systemctl enable zookeeper.service &>> setup.log
+
+# If Zookeeper successfully starts, we don't need this log anymore
+rm setup.log
```
Comment (nathanmarz) [inline rama-cluster/single/start.sh]: What is setup.log for? Also, how does removing it here do anything if the background processes are still outputting to it?

**Review by nathanmarz — CHANGES_REQUESTED** [review]

File: rama-cluster/multi/cluster.tf (hunk 1 above)
Comment (tommy-mor) [reply rama-cluster/multi/cluster.tf]: will remove

File: rama-cluster/single/start.sh (hunk 2 above)
Comment (tommy-mor) [reply rama-cluster/single/start.sh]: i'm not sure, it was there before I messed with it. also I don't think setup.log captures the actual process logs, just the output from enabling the service. usually systemctl enable doesn't output anything

Comment (nathanmarz) [reply rama-cluster/single/start.sh]: Yea, let's remove all of that.

**Comment by nathanmarz** [comment]
Good to merge after those bits of cleanup.

**Pull request merged**